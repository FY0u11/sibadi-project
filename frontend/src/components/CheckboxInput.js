import React from 'react'

const CheckboxInput = ({ name, checked, setChecked, disabled }) => {
  return (
    <div className="form-check form-check-inline">
      <input
        className="form-check-input"
        type="checkBox"
        id={name}
        disabled={disabled}
        checked={checked}
        onChange={() => {
          if (!checked) setChecked(true)
          else setChecked(false)
        }}
      />
      <label className="form-check-label" htmlFor={name}>
        {name}
      </label>
    </div>
  )
}

export default CheckboxInput
