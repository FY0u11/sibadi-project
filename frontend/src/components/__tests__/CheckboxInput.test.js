import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import CheckboxInput from '../CheckboxInput'

afterEach(cleanup)

const props = {
  name: 'test',
  checked: true
}

describe('Testing CheckboxInput component', () => {
  it('It should be properly rendered', async () => {
    const { asFragment } = render(<CheckboxInput {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('It should trigger callback with new checked value on change event', () => {
    const callback = jest.fn(isChecked => {
      props.checked = isChecked
    })
    const { getByRole, rerender } = render(<CheckboxInput {...props} setChecked={callback} />)
    expect(props.checked).toBe(true)
    fireEvent.click(getByRole('checkbox'))
    expect(props.checked).toBe(false)
    rerender(<CheckboxInput {...props} setChecked={callback} />)
    fireEvent.click(getByRole('checkbox'))
    expect(props.checked).toBe(true)
    expect(callback).toBeCalledTimes(2)
  })
})
