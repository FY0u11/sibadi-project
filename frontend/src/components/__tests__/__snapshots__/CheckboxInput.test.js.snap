// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CheckboxInput component It should be properly rendered 1`] = `
<DocumentFragment>
  <div
    class="form-check form-check-inline"
  >
    <input
      checked=""
      class="form-check-input"
      id="test"
      type="checkBox"
    />
    <label
      class="form-check-label"
      for="test"
    >
      test
    </label>
  </div>
</DocumentFragment>
`;
